opentelemetry/src/sdk/trace/tracer.rs

```diff
@@ -286,22 +285,34 @@ impl crate::trace::Tracer for Tracer {
             let status_code = status_code.unwrap_or(StatusCode::Unset);
             let status_message = status_message.unwrap_or(Cow::Borrowed(""));
 
-            SpanData {
-                parent_span_id,
-                span_kind,
-                name,
-                start_time,
-                end_time,
-                attributes,
-                events: events_queue,
-                links,
-                status_code,
-                status_message,
-            }
-        });
-
-        let span_context = SpanContext::new(trace_id, span_id, flags, false, span_trace_state);
-        let mut span = Span::new(span_context, inner, self.clone(), span_limits);
+            let span_context = SpanContext::new(trace_id, span_id, flags, false, trace_state);
```

Comment:
Given that the only differences here are `flags` and `trace_state`, it seems unfortunate to duplicate the calls to `SpanContext::new()` and `Span::new()` entirely.

Reply:
Yeah this one is a bit unfortunate, but it does seem more performant than returning say `let (span_context, inner) = ...` to construct the span from the result.

Reply:
(`~407.23 ns` vs  `~389.59 ns` on my machine for `start-end-span/always-sample` bench.)